Add agent management section to docs

mhausenblas opened the pull request:
This PR addresses https://github.com/open-telemetry/opentelemetry.io/issues/2282

---

**Preview**: https://deploy-preview-2285--opentelemetry.netlify.app/docs/collector/management/

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+weight: 23
+---
+
+Please be sure to review the following documentation:
```
```suggestion
Please be sure to review the following documentation a prerequisite:
```

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+   install the OpenTelemetry collector.
+*  [Configuration][otel-collector-configuration] for how to configure the 
+   OpenTelemetry collector, setting up telemetry pipelines.
+
```
```suggestion
This document describes how you can manage your OpenTelemetry Collector deployment at scale.
```

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+
+## Basics
+
+Telemetry collection at scale requires us to come up with a structured approach
+to manage agents. Typical agent management tasks (ordered from most common to
+least common) include:
```
```suggestion
Telemetry collection at scale requires a structured approach to manage agents.
Typical agent management tasks (ordered from most common to
least common) include:
```

tigrannajaryan replied:
```suggestion
to manage agents. Typical agent management tasks include:
```
I think people may have different opinions about how common the particular task is, so perhaps omit the part in parenthesis.

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+include its version, operating system related information, or capabilities.
+The configuration of the agent refers to its telemetry collection setup,
+for example, the OpenTelemetry collector [configuration][otel-collector-configuration].
+1. Upgrading and downgrading agents and management of agent-specific packages,
+including the base agent functionality and plugins.
```
```suggestion
1. Upgrading/downgrading agents and management of agent-specific packages,
including the base agent functionality and plugins.
```

"the base agent functionality" --> is that mean to be a part of "agent-specific packages"? If so, what's it referring to in particular? Maybe some examples would be helpful. Although IIRC you can't just manage packages in the collector, but need to build a version of the collector with components baked in.

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+handling of TLS certificates (revocation and rotation).
+
+Not every use case requires support for all of the above agent management tasks.
+In the context of OpenTelemetry we aim for a dogfooding, so task 
+_4. Health and performance monitoring_ is ideally done using OpenTelemetry.
```
Hmm. I don't understand this sentence. Dogfooding in my experience is using the software an org (we) produce as a means to better empathize with end users and fix problems that we all experience. How is that relevant here?

mhausenblas replied:
Maybe dogfooding is the wrong term. It's a recommendation to use and open standard (OpenTelemetry? :) for this task. Can drop if you think it's not appropriate here.

svrnm replied:
"OpenTelemetry on OpenTelemetry" or something like that?

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+In the context of OpenTelemetry we aim for a dogfooding, so task 
+_4. Health and performance monitoring_ is ideally done using OpenTelemetry.
+
+## OpAMP
+Observability vendors and cloud providers offer proprietary solutions for 
```
```suggestion

## OpAMP
Observability vendors and cloud providers offer proprietary solutions for 
```

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+agent management. In the open source observability space, there is an emerging standard 
+that you can use for agent management: Open Agent Management Protocol (OpAMP).
+
+The [OpAMP specification][opamp-spec] (status: Beta) defines how to manage a 
+fleet of telemetry data agents. These agents can be  [OpenTelemetry collectors][otel-collector] 
+but you could also be using it for managing other agents, such as Fluent Bit or
+Prometheus in agent mode or indeed a combination of different agents, forming
+a fleet.
```
```suggestion
The [OpAMP specification][opamp-spec] (status: Beta) defines how to manage a 
fleet of telemetry data agents. These agents can be  [OpenTelemetry collectors][otel-collector],
Prometheus in agent mode, Fluent Bit, or other agents in any arbitrary combination.
```

cartermp commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,74 @@
+
+* The **OpAMP server** is part of the control plane and acts as the orchestrator,
+managing a fleet of telemetry agents.
+* The **OpAMP client** us part of the data plane. The client side of OpAMP can
+be implemented in-process, for example, as the case for [OpAMP support in the OpenTelemetry
+collector][opamp-in-otel-collector]. The client side of OpAMP could alternatively
+be implemented out-of-process. For this latter option, you can use a supervisor 
+that takes care of the OpAMP specific communication with the OpAMP server and at 
+the same time controls the telemetry agent, for example to apply a configuration
+or to upgrade it. Note that the supervisor/telemetry communication
+is not part of OpAMP.
```
```suggestion
* The **OpAMP client** is part of the data plane. The client side of OpAMP can
be implemented in-process, for example, as the case for [OpAMP support in the OpenTelemetry
collector][opamp-in-otel-collector]. The client side of OpAMP could alternatively
be implemented out-of-process. For this latter option, you can use a supervisor 
that takes care of the OpAMP specific communication with the OpAMP server and at 
the same time controls the telemetry agent, for example to apply a configuration
or to upgrade it. Note that the supervisor/telemetry communication
is not part of OpAMP.
```

cartermp reviewed (changes requested):
Some minor suggestions.

These docs feel a little weird to me right now -- the second section in particular feels like high-level conceptual docs for _implementers_ of OpAmp, but the primary readers of the docs are end-users. Should that be on the website, or in the OpAmp GitHub repo?

mhausenblas commented:
Thanks for your feedback @cartermp and I'm happy to extend the second section with a usage example based on the Go lib, yes.

austinlparker commented:
In order to end-run the entire 'agent' vs. 'auto-instrumentation' vs. 'whatever' discussion, would it be possible to just include a note that 'agent' is used as a catch-all term for opentelemetry components that respond to OpAMP, or something like it?

chalin reviewed (commented):
Please run Prettier over the file: `npx prettier --ignore-path '' -w <path-to-your-file>`
Thanks!

mviitane commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,97 @@
+standard that you can use for agent management: Open Agent Management Protocol
+(OpAMP).
+
+The [OpAMP specification][opamp-spec] (status: Beta) defines how to manage a
```
You could consider removing the status from this page. It may become outdated when the status changes. The status is anyway visible on the spec page.

mviitane reviewed (approved):
LGTM, this is a very helpful page. Added only one minor comment.

chalin commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,97 @@
+weight: 23
+---
+
+Please be sure to review the following documentation as prerequisite:
```
```suggestion
This document describes how you can manage your OpenTelemetry collector
deployment at scale.

To get the most out of this page you should know how to install and configure
the collector. These topics are covered elsewhere:
```

chalin replied:
Hey @mhausenblas. This had been marked as resolved, but the changes were not applied AFAICT.

chalin replied:
@mhausenblas - btw, in case you didn't know: you can use the "commit" buttons that are a part of the suggestions to accept changes. It makes is easy, and less error prone, to apply changes.

chalin commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,97 @@
+- [Configuration][otel-collector-configuration] for how to configure the
+  OpenTelemetry collector, setting up telemetry pipelines.
+
+This document describes how you can manage your OpenTelemetry collector
+deployment at scale.
+
```
```suggestion
```

chalin replied:
Hey @mhausenblas. This had been marked as resolved, but the changes were not applied AFAICT. (The sentence was moved to the start of the page.

chalin commented on content/en/docs/collector/management.md:
```diff
@@ -0,0 +1,97 @@
+
+Let's have a look at a concrete deployment:
+
+![OpAMP example setup](/img/docs/OTel_OpAMP.svg)
```
I know that you've moved the image already, but I'd like to ask that you relocate the image closer to where it is being used, and rename the file to follow the website's file naming conventions. So move and rename:

- `/img/docs/OTel_OpAMP.svg`

to 

- `/content/en/docs/collector/img/opamp.svg`

Then update the reference in the file.
```suggestion
![OpAMP example setup](../img/opamp.svg)
```

chalin reviewed (changes requested):
Great contribution!

See inline comments for suggestions: an added page description, slightly reworked start-of-page, and image file move.

chalin commented:
Also, if you could rebase, that would be grand!

chalin commented on content/en/docs/collector/management.md:
````diff
@@ -0,0 +1,229 @@
+
+In the `./opamp-go/internal/examples/server` directory, launch the OpAMP server:
+
+```
````
```suggestion
```console
```

chalin commented on content/en/docs/collector/management.md:
````diff
@@ -0,0 +1,229 @@
+First, clone the `open-telemetry/opamp-go` repo:
+
+```sh
+$ git clone https://github.com/open-telemetry/opamp-go.git
````
When there's no command output, and the code-block language is `sh` (or `shell` or `bash`), drop the prompt:

```suggestion
git clone https://github.com/open-telemetry/opamp-go.git
```

chalin commented on content/en/docs/collector/management.md:
````diff
@@ -0,0 +1,229 @@
+Now it's time to launch the supervisor (which in turn will launch your
+OpenTelemetry collector):
+
+```
````
```suggestion
```console
```

chalin commented on content/en/docs/collector/management.md:
````diff
@@ -0,0 +1,229 @@
+You can also query the collector for the metrics exported (note the label
+values):
+
+```
````
```suggestion
```console
```

chalin reviewed (approved):
See inline comments for code-block language fixes.
Other than, that LGTM

mhausenblas commented:
@chalin thanks a bunch for the guidance (and that will be useful for future contribs as well, so really appreciate it!). I've addressed all of the feedback now, I think we're good to merge.
